docuilib/src/components/UILivePreview.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {StyleSheet} from 'react-native';
+import {LiveProvider, LiveEditor} from 'react-live';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import {View, Colors} from 'react-native-ui-lib/core';
+import ReactLiveScope from '../theme/ReactLiveScope';
+
+const messageType = 'LIVE_PREVIEW_CODE_UPDATE_MESSAGE';
+
+export default function UILivePreview({code: codeProp}) {
+  const [code, setCode] = useState(codeProp);
+  const [iframeLoaded, setIframeLoaded] = useState(false);
+  const {siteConfig} = useDocusaurusContext();
+  const iframeRef = useRef(null);
+  const iframeSource = siteConfig?.customFields?.livePreviewSource as string;
+
+  useEffect(() => {
+    if (iframeRef.current && iframeLoaded) {
+      setTimeout(() => {
+        sendMessageToIframe(code);
+      }, 100);
+    }
+  }, [iframeRef, iframeLoaded, code]);
+
+  const sendMessageToIframe = code => {
+    const message = {type: messageType, code};
+    iframeRef.current.contentWindow.postMessage(message, '*');
+  };
+
+  return (
+    <View row gap-s2 style={styles.liveCodeWrapper}>
+      <LiveProvider code={code} scope={ReactLiveScope}>
+        <View flex style={styles.editorWrapper}>
+          <LiveEditor
+            className="font-mono"
+            onChange={setCode}
+            style={{
+              overflowY: 'scroll',
+              scrollbarWidth: 'none' // For Firefox
+            }}
```

Comment:
Lets move to the `styles` (you already have it here)

Reply:
moved into `useMemo`